account_payment_mode/models/account_payment_mode.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+# © 2016 Akretion (Alexis de Lattre <alexis.delattre@akretion.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, fields, api, _
+from openerp.exceptions import ValidationError
+
+
+class AccountPaymentMode(models.Model):
+    """This corresponds to the object payment.mode of v8 with some
+    important changes. It also replaces the object payment.method
+    of the module sale_payment_method of OCA/e-commerce"""
+    _name = "account.payment.mode"
+    _description = 'Payment Modes'
+    _order = 'name'
+
+    name = fields.Char(string='Name', required=True)
+    company_id = fields.Many2one(
+        'res.company', string='Company', required=True, ondelete='restrict',
+        default=lambda self: self.env['res.company']._company_default_get(
+            'account.payment.mode'))
+    bank_account_link = fields.Selection([
+        ('fixed', 'Fixed'),
+        ('variable', 'Variable'),
+        ], string='Link to Bank Account', required=True,
+        help="For payment modes that are always attached to the same bank "
+        "account of your company (such as wire transfer from customers or "
+        "SEPA direct debit from suppliers), select "
+        "'Fixed'. For payment modes that are not always attached to the same "
+        "bank account (such as SEPA Direct debit for customers, wire transfer "
+        "to suppliers), you should choose 'Variable', which means that you "
+        "will select the bank account on the payment order. If your company "
+        "only has one bank account, you should always select 'Fixed'.")
+    fixed_journal_id = fields.Many2one(
+        'account.journal', string='Fixed Bank Journal',
+        domain=[('type', '=', 'bank')], ondelete='restrict')
+    # I need to use the old definition, because I have 2 M2M fields
+    # pointing to account.journal
+    variable_journal_ids = fields.Many2many(
+        'account.journal',
```

Comment:
Although using the full definition, you can put keyword arguments to be more verbose about the arguments (comodel_name, relation, column1 and column2).